app/src/org/commcare/fragments/connect/ConnectPaymentSetupFragment.java

```diff
@@ -0,0 +1,157 @@
+package org.commcare.fragments.connect;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+
+import com.google.android.gms.auth.api.identity.Identity;
+import com.google.android.gms.common.api.ApiException;
+
+import org.commcare.activities.CommCareActivity;
+import org.commcare.connect.ConnectManager;
+import org.commcare.dalvik.R;
+import org.commcare.dalvik.databinding.FragmentConnectPaymentSetupBinding;
+import org.commcare.google.services.analytics.FirebaseAnalyticsUtil;
+import org.commcare.utils.PhoneNumberHelper;
+import org.javarosa.core.services.Logger;
+
+import java.util.Locale;
+
+public class ConnectPaymentSetupFragment extends Fragment {
```

Comment:
There seems to be code in this class which is quite similar to some other connect classes around phone number handling. We need to try very hard to not repeat such blocks of code in the codebase as much as possible.